woocommerce/google-listings-and-ads
# [Don't merge] Add EditFreeCampaign page scaffold

_This is another take on https://github.com/woocommerce/google-listings-and-ads/pull/226 and realizes 1. and half of 2. of the plan from https://github.com/woocommerce/google-listings-and-ads/pull/226#issuecomment-790215574_
### Changes proposed in this Pull Request:

- Add edit free campaign page.
- Link it from Edit buttons on Dashboard.
- Workaround full container/pane view.
- Scaffold UI.
- Move non-problematic, re-usable Setup components to a shared `.~/components` folder.

Implements part of https://github.com/woocommerce/google-listings-and-ads/issues/156.

:warning: There is a number of issues that will be covered in separate PRs not to blur the picture, and make them easier to merge. (see "To be implemented..." section below



### Screenshots:



### Detailed test instructions:

0. Check that the project builds (dependencies are resolved correctly for Onboarding)
1. Go to [the GLA Dashboard](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard)
2. Click the "edit" button on a free listing
3. Check if the form on both steps matches the [Figma designs](https://www.figma.com/file/jZUpa8eTrnrK1Lwt2ry7zk/Native-Google-Integration?node-id=4162%3A0) in high-level (looks exactly as in Onboarding Setup)

### Notes

I made "< Edit free listings" in different size and color than it is defined in [Figma](https://www.figma.com/file/jZUpa8eTrnrK1Lwt2ry7zk/Native-Google-Integration?node-id=4162%3A0), as I thought it should share the look with other pages: it is like the one in onboarding.

### To be implemented in follow-up PRs
- [ ] Make step header configurable in `<Hero>` component. "STEP THREE"-> "STEP TWO"
- [ ] Remove autosave from Shipping Rate and Shipping Time
- [ ] Make the `<TopBar>` a component shared by Onboarding and EditCampaign
- [ ] Highlight edited program once finished
- [ ] Saving on 2nd step save button
- [ ] Add prompt on leave
- [ ] Add tracking events

### Doubts:

1. I don't like the way `FullContainer` is implemented. It more or less follows `FullPage`. However, it feels to me like a quite unstable hack around WooCommerce's navigation and layout. That solution relies on implementation details of various components in WC-admin, and just overlay, or hide them with CSS. Also, it gives a false impression that it has anything to do with the content inside.

   `<Full*><Foo></Full*>` works as well as `<Full* /><Foo />`, `<Foo><Full* /></Foo>`, or even `<Foo />` + `<Full* />` placed in some distant part of DOM.

   I think we could either use WC-admins API, navigation, and layout correctly or change it so it could be used the way we need without hacks.
